sonic_platform_base/psu_base.py

```diff
@@ -0,0 +1,99 @@
+#
+# psu_base.py
+#
+# Abstract base class for implementing a platform-specific class with which
+# to interact with a power supply unit (PSU) in SONiC
+#
+
+try:
+    import abc
+    from . import device_base
+except ImportError as e:
+    raise ImportError(str(e) + " - required module not found")
+
+
+class PsuBase(device_base.DeviceBase):
+    """
+    Abstract base class for interfacing with a power supply unit
+    """
+    __metaclass__ = abc.ABCMeta
+
+    # Possible fan directions
+    FAN_DIRECTION_INTAKE = "intake"
+    FAN_DIRECTION_EXHAUST = "exhaust"
+
+    # Possible fan status LED colors
+    STATUS_LED_COLOR_GREEN = "green"
+    STATUS_LED_COLOR_RED = "red"
+    STATUS_LED_COLOR_OFF = "off"
+
+    @abc.abstractmethod
```

Comment:
One suggestion:  handle the PSU fans in chassis class just like other fans. 
And is it possible that one PSU installed with multi fans?